tests/integration/azure/test_azure_credentials.py

```diff
@@ -0,0 +1,142 @@
+import base64
+import re
+from unittest.mock import MagicMock
+
+import pytest
+from databricks.labs.blueprint.tui import MockPrompts
+
+from databricks.labs.ucx.assessment.azure import AzureServicePrincipalInfo
+from databricks.labs.ucx.assessment.crawlers import _SECRET_PATTERN
+from databricks.labs.ucx.azure.access import StoragePermissionMapping
+from databricks.labs.ucx.azure.azure_credentials import AzureServicePrincipalMigration
+
+
+@pytest.fixture
+def prepare_spn_migration_test(ws, debug_env, make_random):
+    def inner(read_only=False) -> dict:
+        spark_conf = ws.clusters.get(cluster_id=debug_env["TEST_LEGACY_SPN_CLUSTER_ID"]).spark_conf
+
+        application_id = spark_conf.get("fs.azure.account.oauth2.client.id")
+
+        secret_matched = re.search(_SECRET_PATTERN, spark_conf.get("fs.azure.account.oauth2.client.secret"))
+        if secret_matched:
+            secret_scope, secret_key = (
+                secret_matched.group(1).split("/")[1],
+                secret_matched.group(1).split("/")[2],
+            )
+        assert secret_scope is not None
+        assert secret_key is not None
+
+        secret_response = ws.secrets.get_secret(secret_scope, secret_key)
+        client_secret = base64.b64decode(secret_response.value).decode("utf-8")
+
+        end_point = spark_conf.get("fs.azure.account.oauth2.client.endpoint")
+        directory_id = end_point.split("/")[3]
+
+        name = f"testinfra_storageaccess_{make_random(4).lower()}"
+
+        azure_resource_permissions = MagicMock()
+        azure_resource_permissions.load.return_value = [
+            StoragePermissionMapping(
+                prefix="abfss://things@labsazurethings.dfs.core.windows.net/avoid_ext_loc_overlap",
```

Comment:
There should be an env var for this

Reply:
No there is no env var for this, please add one, so this code as well as the [test_access.py](https://github.com/databrickslabs/ucx/blob/f951aac01cecbf70584fb71ef59f06cedd08a6f0/tests/integration/azure/test_access.py#L19C2-L33C92) can use it